### ISSUE TITLE
Is there a complete list of metadata format options?

### ISSUE THREAD

**Issue by jefferysmall** [altdesktop/playerctl#150]
I just started using playerctl and love it.  I'm wondering if the full set of possible format options is available anywhere?  These seem to come in three classes:

Internal:  playerName, duration, etc.

Xesam:  xesam:title, xesam:artist, etc.

Mpris: mpris:length

Thanks for any poointers

**Comment by acrisci** [comment]
I should add this to the documentation formally. The internal ones are currently not documented anywhere except the examples.

As for the xesam, the xesam spec is no longer maintained. There used to be a spec that tried to give formal names to metadata concepts to prevent semantic drift but I can't find it anywhere. So it's just kind of cruft now.

The mpris: ones are in the [official mpris spec](https://specifications.freedesktop.org/mpris-spec/latest/) and some of them are required to be there. I think length is one of those and maybe a track.

Anyway this definitely deserves something written out, so I'll do that soon.

**Comment by acrisci** [comment]
But to actually answer this question, use a plain metadata command and all of the keys you see can be used as template variables.

```
playerctl metadata
```

**Comment by jefferysmall** [comment]
Thanks for the tip -- it works.  Before I posted I tried a couple of things like:

playerctl metadata --format

And got errors.  I guess I just didn't try long and hard enough!  :-) 

While we're discussing the specs, I see that there is a user rating which I take to be a percentage out of 1.  But specifically, I have a 4/5 star rating in Clementine for a given song, but I'm seeing:

clementine xesam:userRating          0.80000001192092896

Does the trailing stuff (0000001192092896) tell us anything?
